docs/decisions/0013-backend-for-frontend-api.md

```diff
@@ -0,0 +1,69 @@
+# 0013. Introduction of the backend-for-frontend orchaestration API
+
+## Status
+
+Accepted (December 2024)
+
+## Context
+
+As part of supporting the feature of auto enrollment of default enrollment intentions ([see ADR](https://github.com/openedx/edx-enterprise/blob/master/docs/decisions/0015-default-enrollments.rst))
+and auto-apply of subscription plan licenses, it was determined the sequencing of API calls to resolve in the existing 
+frontend root loader would result in a new set of waterfall API calls on top of existing reconciliation of the current status of
+the user's subsidies when arriving on the enterprise learner portal. Furthermore, in support of including this feature,
```

Comment:
Bit of a correction here: the primary concern is the interaction of API requests between the `rootLoader` and `dashboardLoader`, as they operate asynchronously in parallel. The challenge presents itself in that the side effect of realizing default enrollment intentions would need to happen in the `rootLoader` after processing the auto-applied subscription license, which would effectively change the response of the enrollments retrieved by the `dashboardLoader`.

Another way of putting it might be that, currently the `rootLoader` and `dashboardLoader` operate independently as neither mutates the data for the other. To support auto-enrollment, though, the `rootLoader` would mutate the data retrieved by `dashboardLoader`, which would lead to additional complexities around data orchestration regarding optimistic cache updates and possible race conditions.